protocol_details/rpmsg_protocol.rst

```diff
@@ -0,0 +1,89 @@
+.. _rpmsg-protocol-work-label:
+
+=====================
+RPMsg Protocol Layers
+=====================
+
+The whole communication implementation can be separated in three different ISO/OSI layers - Transport, Media Access Control and Physical layer. Each of them can be implemented separately and for example multiple implementations of the Transport Layer can share the same implementation of the MAC Layer (VirtIO) and the Physical Layer. Each layer is described in following sections.
+
+.. image:: ../images/protocol_layers.jpg
+
+Physical Layer – Shared Memory
+------------------------------
+
+The solution proposed in this document requires only two basic hardware components - shared memory (accessible by both communicating sides) and inter-core interrupts (in a specific configuration optional). The minimum configuration requires one interrupt line per communicating core meaning two interrupts in total. This configuration is briefly presented in figure at the beginning of this section. It is to be noticed that no inter-core synchronization hardware element such as inter-core semaphore, inter-core queue or inter-core mutex is needed! This is thanks to the nature of the `virtqueue <https://docs.oasis-open.org/virtio/virtio/v1.3/csd01/virtio-v1.3-csd01.html#x1-270006>`_, which uses single-writer-single-reader circular buffering. (As defined in next subsection).
```

Comment:
Rephrase "in a specific configuration optional" as "optional in a specific configuration"?